packages/musap-rn-kms/src/agent/MusapKeyManagerSystem.ts

```diff
@@ -0,0 +1,114 @@
+import { IKey, ManagedKeyInfo, MinimalImportableKey, TKeyType } from '@veramo/core';
+import {
+  KeyAlgorithmType,
+  KeyGenReq,
+  MusapKey,
+  MusapModuleType, signatureAlgorithmFromKeyAlgorithm,
+  SignatureAlgorithmType,
+  SignatureFormat,
+  SignatureReq,
+} from '@sphereon/musap-react-native'
+import { SscdType } from '@sphereon/musap-react-native/src/types/musap-types';
+import { KeyManagementSystem } from '@veramo/kms-local';
+import { AbstractPrivateKeyStore } from '@veramo/key-manager';
+import 'react-native-get-random-values'
+import { v4 as uuid } from 'uuid';
+import { TextDecoder } from 'text-encoding';
+
+export class MusapKeyManagementSystem extends KeyManagementSystem {
+  private musapKeyStore: MusapModuleType;
+
+  constructor(keyStore: MusapModuleType) {
+    super(keyStore as unknown as AbstractPrivateKeyStore);
+    this.musapKeyStore = keyStore;
+    this.musapKeyStore.enableSscd('TEE')
+  }
+
+  async listKeys(): Promise<ManagedKeyInfo[]> {
+    try {
+      const keysJson: MusapKey[] = (await this.musapKeyStore.listKeys()) as MusapKey[];
+      return keysJson.map((key) => this.asMusapKeyInfo(key));
+    } catch (error) {
+      console.error('Failed to list keys:', error);
+      throw error;
+    }
+  }
+
+  async createKey(args: { type: TKeyType; sscdType?: SscdType }): Promise<ManagedKeyInfo> {
+    const sscdType: SscdType = args.sscdType ? args.sscdType : 'TEE';
+    const keyGenReq: KeyGenReq = {
+      keyAlgorithm: args.type as KeyAlgorithmType,
```

Comment:
This is a cheat.

You are taking TKeyType as arg
````
async createKey(args: { type: TKeyType
````
Then you cast to to KeyAlgorithmType
````
TKeyType is
export type TKeyType = 'Ed25519' | 'Secp256k1' | 'Secp256r1' | 'X25519' | 'Bls12381G1' | 'Bls12381G2' | 'RSA'
````

KeyAlgorithmType is
````
export type KeyAlgorithmType =
    | 'RSA1K'
    | 'RSA2K'
    | 'RSA4K'
    | 'ECCP256K1'
    | 'ECCP256R1'
    | 'ECCP384K1'
    | 'ECCP384R1'
````
In the test we have
````
        // @ts-ignore
        const keyManagedInfo = await kms.createKey({type: 'ECCP256R1'})
````        
That's also cheating. you'll need to create a mapper function for this.
We don't support all entries in both directions, so the mapper function needs to throw an unsupported error for unmappable entries.
RSA should map to RSA2K for when the MUSAP library supports it.

Reply:
added a mapper function for it